src/CircleEditor/CircleEditorDocument.ts

```diff
@@ -191,4 +195,333 @@ export class CircleEditorDocument extends Disposable implements vscode.CustomDoc
       }
     };
   }
+
+  /**
+   * Guess data's type (int or float)
+   */
+  private guessExactType(n: any) {
+    if (Number(n) % 1 === 0) {
+      return "int";
+    }
+    else if (Number(n) % 1 !== 0) {
+      return "float";
+    }
+  }
+
+  /**
+   * Encode key and value entered in the calculator using flexbuffers.
+   * Send encoded data.
+   */
+  public sendEncodingData(message: any) {
+    let fbb = flexbuffers.builder();
+    fbb.startMap();
+    for (const key in message.data) {
+      fbb.addKey(key);
+      const val = message.data[key][0];
+      const valType = message.data[key][1];
+      if (valType === "boolean") {
+        if (val === "true" || val === true) {
+          fbb.add(true);
+        }
+        else if (val === "false" || val === false) {
+          fbb.add(false);
+        }
+        else {
+          Balloon.error("'boolean' type must be 'true' or 'false'.", false);
+          return;
+        }
+      }
+      else if (valType === "int") {
+        if (this.guessExactType(val) === 'float') {
+          Balloon.error("'int' type doesn't include decimal point.", false);
+          return;
+        }
+        fbb.addInt(Number(val));
+      }
+      else {
+        fbb.add(String(val));
+      }
+    }
+    fbb.end();
+    const res = fbb.finish();
+    const buf = Buffer.alloc(res.byteLength);
+    const view = new Uint8Array(res);
+    for (let i = 0; i < buf.length; ++i) {
+      buf[i] = view[i];
+    }
+    const data = Array.from(buf);
+    let responseData = {
+      command: 'responseEncodingData',
+      data: data
+    };
+    this._onDidChangeContent.fire(responseData);
+  }
+
+  /**
+   * Decode customOptions using flexbuffers when we click custom operator's node.
+   * Send value's type of custom operator. 
+   */
+  public sendCustomType(message: any) {
+    const msgData: any = message.data;
+    const subgraphIdx: number = msgData._subgraphIdx;
+    const operatorIdx: number = msgData._nodeIdx;
+    const target = this._model.subgraphs[subgraphIdx].operators[operatorIdx].customOptions;
+    const buffer = Buffer.from(target);
+    const ab = new ArrayBuffer(buffer.length);
+    const view = new Uint8Array(ab);
+    for (let i = 0; i < buffer.length; ++i) {
+      view[i] = buffer[i];
+    }
```

Comment:
Maybe not used?

Reply:
which means that writes to any ArrayBufferView will write through to the original Buffer instance.

I refered below url.
reference : <https://stackoverflow.com/questions/8609289/convert-a-binary-nodejs-buffer-to-javascript-arraybuffer>

Reply:
I mean `view` variable is not used anywhere...?

Reply:
I am not sure about this code as I do not write this, but as I searched, we cannot write data directly into ArrayBuffer.
We can use 'view' object to write something into ArrayBuffer. So with those lines we indirectly store data into ab ArrayBuffer object.
https://javascript.info/arraybuffer-binary-arrays

He tries to make 'target' (which is a number array) into ArrayBuffer type, so that he can use flexbuffers.toObject method. I want to change this part if there is a better way.